src/modules/flow/si114x/si114x.c

```diff
@@ -0,0 +1,370 @@
+    data = 0xff;
+    if (!sol_i2c_write_register(mdata->context, REG_MEAS_RATE0, &data, 1)) {
+        SOL_WRN("Couldn't set measure speed");
+        return false;
+    }
+
+    // set autorun
+    data = CMD_ALS_AUTO;
+    if (!sol_i2c_write_register(mdata->context, REG_COMMAND, &data, 1)) {
+        SOL_WRN("Couldn't set mode to auto-run");
+        return false;
+    }
+    mdata->fully_initialized = true;
+    return false;
+}
+
+static bool
+setup_device_pt2(void *d)
+{
+    struct si114x_data *mdata = (struct si114x_data *)d;
+    uint8_t data = SI114X_HW_KEY;
+
+    if (!sol_i2c_write_register(mdata->context, REG_HW_KEY, &data, 1)) {
+        SOL_WRN("Couldn't set the REG_HW_KEY to SI114X_HW_KEY");
+        return false;
+    }
+    timer_reschedule(mdata, 100, &setup_device_pt3, mdata);
+    return false;
+}
+
+static bool
+setup_device(void *d)
+{
+    struct si114x_data *mdata = (struct si114x_data *)d;
+    uint8_t data;
+
+    // Initialize the device to an empty state
+    // zero out measuring rate
+    data = 0;
+    if (!sol_i2c_write_register(mdata->context, REG_MEAS_RATE0, &data, 1)) {
+        SOL_WRN("Couldn't reset the REG_MEAS_RATE0 register");
+        return false;
+    }
+
+    if (!sol_i2c_write_register(mdata->context, REG_MEAS_RATE1, &data, 1)) {
+        SOL_WRN("Couldn't reset the REG_MEAS_RATE1 register");
+        return false;
+    }
+
+    // disable IRQ MODES
+    // these are undocumented in the datasheet, but mentioned in Adafruit's code
+    if (!sol_i2c_write_register(mdata->context, REG_IRQ_MODE1, &data, 1)) {
+        SOL_WRN("Couldn't reset the REG_IRQ_MODE1 register");
+        return false;
+    }
+
+    if (!sol_i2c_write_register(mdata->context, REG_IRQ_MODE2, &data, 1)) {
+        SOL_WRN("Couldn't reset the REG_IRQ_MODE2 register");
+        return false;
+    }
+
+    // turn off interrupts
+    if (!sol_i2c_write_register(mdata->context, REG_INT_CFG, &data, 1)) {
+        SOL_WRN("Couldn't reset the REG_INT_CFG register");
+        return false;
+    }
+
+    data = 0xff;
+    if (!sol_i2c_write_register(mdata->context, REG_IRQ_STATUS, &data, 1)) {
+        SOL_WRN("Couldn't reset the REG_IRQ_STATUS register");
+        return false;
+    }
+
+    // send a reset
+    data = CMD_RESET;
+    if (!sol_i2c_write_register(mdata->context, REG_COMMAND, &data, 1)) {
+        SOL_WRN("Couldn't reset the REG_COMMAND register");
+        return false;
+    }
+
+    timer_reschedule(mdata, 100, &setup_device_pt2, mdata);
+    return false;
+}
+
+static int
+si114x_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    struct si114x_data *mdata = data;
+
+    SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options, SOL_FLOW_NODE_TYPE_LIGHT_SENSOR_SI114X_OPTIONS_API_VERSION, -EINVAL);
+
+    mdata->fully_initialized = false;
+    mdata->opts = (struct sol_flow_node_type_light_sensor_si114x_options *)options;
+    mdata->uv_calibration.data_raw = mdata->opts->calibration.val;
```

Comment:
datasheet uses uchar values, we use int32 - that's the only reason I used it like this.
I wanted to be able to do something like "options = INIT_UV | INIT_IR" on the config one day.
if I add all 4 necessary parameters on the config this will get too big:
_(si114x:first_register=0x1,second_register=0x4,third=register=0x5,fourth_register=0x10) OUT

Too, too big.